src/instructlab/sdg/configs/skills/data_recipe/default_recipe.yaml

```diff
@@ -0,0 +1,3 @@
+datasets:
+  - path: instructlab/InstructLabCommunity
```

Comment:
This worries me a bit ... downloading stuff from Hugging Face, down in the depths of the library

UX-wise, it feels a little like 'ilab model download' ... something you would do as part of setting up your workflow

It's like, you could plausibly do `data generate` and `model train` iterations on your laptop on a plane with no Internet, except for this?

Consider also a unit test like #208 ... all of this datamixing code is self-contained, apart from this external dependency on Hugging Face?

If we had some sort if `ilab data setup` command ... and it downloaded this dataset into local cache ... what else might that command do? e.g. validate that the teacher models required are available from the configured OpenAI endpoint?

I'm not really suggesting this as a blocker for merging ... but I would like to file an issue which acknowledges that this isn't ideal and has some sort of plan for resolving it

Reply:
I agree and share some of these concerns, as it's a download that just happens behind the scenes automatically. I know some downstream users intend to replace this HuggingFace with a file on disk as the preconfigured dataset, which is why the code supports loading from either. It's a 32MB file, which is not necessarily too large to just include as part of our python package, although that may not be ideal either.

I'd prefer to not introduce a new command for this right now, as depending on how we address this it may not require a separate command to download (like if we just ship it in some python package). Or, it may need a separate command. Either way I think a proper fix warrants a longer discussion than what we'll want for this release.

Is there a better alternative we should go with for this release? Do you think just placing the file directly in the git repository and including it as part of the SDG package is a less bad option?

Reply:
A 32Mb file is not ideal, but in the context of some of these other packages ...

Paging @tiran @dhellmann @n1hility

Reply:
> download that just happens behind the scenes automatically

To a security engineer, this is a horror story in eight words. CC:  @sidhpurwala-huzaifa 

How is the download triggered? How is the content tracked, verified and signed? Please assume that $user wants to run InstructLab on an air-gapped system, with all data pre-loaded and without any access to the internet. Some users want to use InstructLab with PII or confidential data.

Very recently we had a similar behavior in a 3rd party dependency. It triggered a HTTPS request and a download every time we ran `ilab` command. The code ran on `import` time.

I would prefer a Python package or another mechanism to deliver and track the file.

Reply:
Generally, libraries should not introduce side-effects like downloading files and writing to potentially protected locations on disk.

More specifically, when we get to the point of running tests for individual packages before using them downstream, those tests will have to run without accessing the internet. Everything they need will have to be installed before the tests run.

Reply:
This entire discussion makes sense. Are there suggestions for how we do want to ship this precomputed dataset in upstream? How we want to reference it? And, how would downstream users supply their own precomputed dataset (replacing/obviating the upstream one specified here) in their own releases?

Reply:
These aren't the only inputs the system needs, right? Is there a pattern to follow in how things like models are made available for users?

Reply:
See https://github.com/instructlab/sdg/pull/203#issuecomment-2250444499